File: src/designpatterns/decorator/DecoratorTS/beverage.ts
/** ABSTRACT/INTERFACE COMPONENT
 * This is the thing that Decorator targets to 
 * decorate (by extending).
 */
abstract class Beverage {
    abstract cost(): number;
    abstract description(): string;
}

/** CONCRETE COMPONENT
 * define 'hard-coded' Component for the Decorator.
 */
class DarkRoast extends Beverage {
    cost(): number { return 3.45; }

    description(): string { return "Dark Roast"; }
}

class LightRoast extends Beverage {
    cost(): number { return 3.45; }

    description(): string { return "Light Roast"; }
}

export { Beverage, DarkRoast, LightRoast };

File: src/designpatterns/decorator/DecoratorJava/Beverage.java
package designpatterns.decorator.DecoratorJava;

public abstract class Beverage {

    abstract double cost();
    abstract String description();
}

class DarkRoast extends Beverage {
    
    @Override
    double cost() { return 3.45; }

    @Override
    String description() { return "Dark Roast"; }
}

class LightRoast extends Beverage {

    @Override
    double cost() { return 3.45; }

    @Override
    String description() { return "Light Roast"; }
}